apps/web/utils/configurationHelper.ts

```diff
@@ -0,0 +1,109 @@
+import axios from 'axios';
+import https from 'https';
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+
+const envFilePath = path.resolve(__dirname, "../.env");
+const envTmpFilePath = path.resolve(__dirname, '../.env.tmp');
+
+// read .env file & convert to array
+const readEnvVars = (path: string) => fs.readFileSync(path, "utf-8").split(os.EOL);
+
+const getEnvValue = (path: string, key: string) => {
+  // find the line that contains the key (exact match)
+  const matchedLine = readEnvVars(path).find((line) => line.split("=")[0] === key);
+  // split the line (delimiter is '=') and return the item at index 2
+  return matchedLine !== undefined ? matchedLine.split("=")[1] : null;
+};
+
+const setEnvValue = (path: string, key: string, value: string) => {
+  const envVars = readEnvVars(path);
+  const targetLine = envVars.find((line) => line.split("=")[0] === key);
+  if (targetLine !== undefined) {
+    // update existing line
+    const targetLineIndex = envVars.indexOf(targetLine);
+    // replace the key/value with the new value
+    envVars.splice(targetLineIndex, 1, `${key.toUpperCase()}="${value}"`);
+  } else {
+    // create new key value
+    envVars.push(`${key.toUpperCase()}="${value}"`);
+  }
+  // write everything back to the file system
+  fs.writeFileSync(path, envVars.join(os.EOL));
+};
+
+const getConfiguration = async () => {
+  const requiredEnvVars = ['API_ENDPOINT', 'API_SECURITY_TOKEN', 'CONFIG_ID'];
+  let requiredEnvData: string = '';
+
+  console.log('Verifying required env variables...');
+
+  for (let i = 0; i < requiredEnvVars.length; i++) {
+    const envVar = requiredEnvVars[i];
+    if (!getEnvValue(envFilePath, envVar)) {
+      console.log(`Missing or invalid required environment variable: ${envVar}`);
+      return;
+    }
+
+    const envValue = process.env[envVar];
+    requiredEnvData += `${envVar}=${envValue}`;
+
+    // For last iteration just skip adding
+    if (i !== requiredEnvVars.length - 1) {
+      requiredEnvData += '\n';
+    }
+  }
+
+  console.log('Required env variables found ✅');
+
+  // Create .env.tmp file
+  console.log('Creating .env.tmp file...');
```

Comment:
I would only have one console log that confirms writing the env file.
And only log errors and use some tryCatch around the operations.